src/v/cluster/topic_recovery_validator.cc

```diff
@@ -0,0 +1,376 @@
+            // external service issue (note that the manifest path will end in
+            // .bin but the value is just a hint of the HEAD request that
+            // generated the abnormal result)
+            vlog(
+              clusterlog.error,
+              "{}/{} ({}) manifest NOT found. download_result: {}",
+              ntp,
+              rev_id,
+              cloud_storage::generate_partition_manifest_path(ntp, rev_id, {}),
+              download_res);
+
+            return download_issue;
+        });
+}
+
+auto do_validate_manifest_metadata(
+  cloud_storage_clients::bucket_name const& bucket,
+  model::ntp const& ntp,
+  model::initial_revision_id rev_id,
+  cloud_storage::remote& remote,
+  retry_chain_node& rtc,
+  recovery_checks checks) -> ss::future<validation_result> {
+    using enum validation_result;
+    // delegate to anomalies_detector to run few checks on the remote partition,
+    // interpret the result to produce a validation_result
+
+    // for logging, note that `path` is only a hint, we could be dealing with a
+    // .json manifest instead (but at v24.1 it's a remote possibility)
+    auto get_path = [&] {
+        return cloud_storage::generate_partition_manifest_path(
+          ntp, rev_id, cloud_storage::manifest_format::serde);
+    };
+
+    auto rtc_logger = retry_chain_logger{
+      clusterlog, rtc, ssx::sformat("{}", ntp)};
+    // allow enough quota just to check existance and self metadata check
+    // and few segments existance
+    // TODO ensure that future evolution of anomalies detector operate on a
+    // "metadata first" behaviour
+    auto detector = cloud_storage::anomalies_detector{
+      std::move(bucket),
+      ntp,
+      rev_id,
+      remote,
+      rtc_logger,
+      rtc.root_abort_source()};
+
+    // spin up detector
+    auto anomalies_fut = co_await ss::coroutine::as_future(detector.run(
+      rtc,
+      cloud_storage::anomalies_detector::segment_depth_t{
+        checks.max_segment_depth}));
+
+    if (anomalies_fut.failed()) {
+        // propagate shutdown exceptions, but treat other exceptions as hard
+        // failures. likely could be serde exceptions for a corrupt file
+        auto ex = anomalies_fut.get_exception();
+        if (ssx::is_shutdown_exception(ex)) {
+            co_return ss::coroutine::exception(ex);
+        }
+        vlog(
+          clusterlog.error,
+          "{}/{} ({}) manifest metadata check threw exception: {}",
+          ntp,
+          rev_id,
+          get_path(),
+          ex);
+        co_return failure;
+    }
+
+    // examine anomalies, some are real failures while other can be
+    // tollerated
+    auto anomalies = anomalies_fut.get();
+    if (anomalies.status == cloud_storage::scrub_status::failed) {
+        // download issue. could be a configuration issue or a s3 issue
+        vlog(
+          clusterlog.error,
+          "{}/{} ({}) manifest metadata check download failure",
```

Comment:
nit: rather than repeating {}/{} everywhere, consider using a retry_chain_logger? It's already instantiated up there

Reply:
Also, I think it's worth considering putting all of these methods into their own `topic_recovery_validator` class, that can own common members (ntp, revision, loggers, paths, etc). May also make this more easily unit testable, and would have a great place (in the header) to detail expectations about the return values of these functions

Reply:
done